src/analyze/analyze.c

```diff
@@ -213,10 +213,10 @@ static int compare_unit_start(const UnitTimes *a, const UnitTimes *b) {
         return CMP(a->activating, b->activating);
 }
 
-static void unit_times_free(UnitTimes *t) {
-        for (UnitTimes *p = t; p->has_data; p++)
+static UnitTimes* unit_times_free(UnitTimes *t) {
```

Comment:
maybe add a comment here, that this destroy the whole frickin' array. The function is called so innocently, one might not expect that